### PR TITLE
Verify proofs on app client.

### PR DESCRIPTION
**Pull request by aterentic-ethernal**

### CODE REVIEW

File: src/sync.rs [hunk 1]
```diff
@@ -149,6 +153,31 @@ pub async fn sync_block_headers(
 							count.to_be_bytes(),
 						)
 						.unwrap();
+
+					// Push the randomly selected cells to IPFS
```
Comment (sh3ll3x3c) [inline src/sync.rs:157]: Are we pushing just the RPC retrieved cells or all of them?

File: src/app_client.rs [hunk 2]
```diff
@@ -51,10 +56,55 @@ async fn process_block(
...
+			log::info!(
+				"Completed {} verification rounds for block {}",
+				count,
+				block.number
+			);
+
+			// TODO: If there are some invalid cells should we fail?
+
+			for cell in cells {
+				if let Err(error) = ipfs.insert(cell.clone().to_ipfs_block()) {
```
Comment (sh3ll3x3c) [inline src/app_client.rs:86]: Same as with sync, we should make sure that we're sending just the RPC retrieved cells, as not to spam the network with duplicates...

File: src/app_client.rs [hunk 3]
```diff
@@ -51,10 +56,55 @@ async fn process_block(
...
+			cells.extend(rpc_fetched);
+
+			if positions.len() > cells.len() {
+				return Err(anyhow!(
+					"Failed to fetch {} cells",
+					positions.len() - cells.len()
+				));
+			}
+
+			let count = verify_proof(
```
Comment (sh3ll3x3c) [inline src/app_client.rs:70]: @prabal-banerjee for cells belonging to the same appID (sequentially ordered), does it make sense at all to verify proof, or should the app client just check the block confidence and then proceed with the reconstruction?

File: src/app_client.rs (hunk 2 above)
Comment (aterentic-ethernal) [reply src/app_client.rs:86]: Sure Ill change that, on both places.

File: src/app_client.rs (hunk 3 above)
Comment (sh3ll3x3c) [reply src/app_client.rs:70]: on the other hand, it does make sense to verify the proof, but the sequential cell proof verification mustn't be used for confidence score. So we should be good here...

File: src/app_client.rs (hunk 2 above)
Comment (aterentic-ethernal) [reply src/app_client.rs:86]: On three places actually.

**Review by sh3ll3x3c — APPROVED** [review]

File: src/app_client.rs (hunk 3 above)
Comment (prabal-banerjee) [reply src/app_client.rs:70]: IMO, the app clients should verify all downloaded cells. The confidence score should be just computed on the LC and not the app client.

**Pull request merged**